.github/workflows/k8s-e2e.yml

```diff
@@ -25,115 +26,22 @@ jobs:
         with:
           go-version-file: 'go.mod'
           cache-dependency-path: "go.sum"
-      - name: Setup Kind
```

Comment:
All of this logic, currently tied into github actions / engineerd/setup-kind, is making it (quite) difficult to test locally.

With this proposed PR, the logic gets moved to an independent shell script that should make it much easier to test locally, and also to control what version of what gets used.